source/module_hsolver/kernels/cuda/math_kernel_op.cu

```diff
@@ -117,14 +128,35 @@ __global__ void line_minimize_with_block(
     __syncthreads();
 
     // just do some parallel reduction in shared memory
-    for (int ii = THREAD_PER_BLOCK >> 1; ii > 0; ii >>= 1) {
+    for (int ii = THREAD_PER_BLOCK >> 1; ii > WARP_SIZE; ii >>= 1) {
         if (tid < ii) {
             data[tid] += data[tid + ii];
             data[THREAD_PER_BLOCK + tid] += data[THREAD_PER_BLOCK + tid + ii];
             data[2 * THREAD_PER_BLOCK + tid] += data[2 * THREAD_PER_BLOCK + tid + ii];
         }
         __syncthreads();
     }
+    if (tid < WARP_SIZE) {
```

Comment:
In this PR, you have optimized the performance of this code, please add annotations to explain code below.